### ISSUE TITLE
Unable to compile IETF version

### ISSUE THREAD

**Issue by ronaldtse** [CalConnect/csd-icalendar-series#1]
As reported by @douglm . This still occurs after updating to the latest gems.

```
FILENAME=draft-ietf-calext-icalendar-series.adoc; \
	echo "Compiling locally..."; bundle exec metanorma $FILENAME
Compiling locally...
[metanorma] Error: nits format is not supported for this standard.
Reading workgroups from https://tools.ietf.org/wg/...
Reading workgroups from https://irtf.org/groups...
bundler: failed to load command: metanorma (/Users/me/.rbenv/versions/2.6.5/bin/metanorma)
TypeError: no implicit conversion of Array into String
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/nokogiri-1.10.4/lib/nokogiri/xml/node.rb:476:in `in_context'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/nokogiri-1.10.4/lib/nokogiri/xml/node.rb:476:in `parse'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/nokogiri-1.10.4/lib/nokogiri/xml/document_fragment.rb:19:in `initialize'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/nokogiri-1.10.4/lib/nokogiri/xml/document.rb:235:in `new'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/nokogiri-1.10.4/lib/nokogiri/xml/document.rb:235:in `fragment'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/nokogiri-1.10.4/lib/nokogiri/xml/builder.rb:354:in `<<'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-ietf-1.0.6/lib/asciidoctor/rfc/v2/blocks.rb:227:in `block (4 levels) in example'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/nokogiri-1.10.4/lib/nokogiri/xml/builder.rb:394:in `insert'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/nokogiri-1.10.4/lib/nokogiri/xml/builder.rb:377:in `method_missing'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-ietf-1.0.6/lib/asciidoctor/rfc/v2/blocks.rb:226:in `block (3 levels) in example'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-ietf-1.0.6/lib/asciidoctor/rfc/v2/blocks.rb:214:in `each'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-ietf-1.0.6/lib/asciidoctor/rfc/v2/blocks.rb:214:in `block (2 levels) in example'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/nokogiri-1.10.4/lib/nokogiri/xml/builder.rb:394:in `insert'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/nokogiri-1.10.4/lib/nokogiri/xml/builder.rb:377:in `method_missing'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-ietf-1.0.6/lib/asciidoctor/rfc/v2/blocks.rb:213:in `block in example'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/nokogiri-1.10.4/lib/nokogiri/xml/builder.rb:295:in `initialize'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/nokogiri-1.10.4/lib/nokogiri/xml/builder.rb:254:in `new'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/nokogiri-1.10.4/lib/nokogiri/xml/builder.rb:254:in `with'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-ietf-1.0.6/lib/asciidoctor/rfc/common/base.rb:355:in `noko'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-ietf-1.0.6/lib/asciidoctor/rfc/v2/blocks.rb:212:in `example'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-ietf-1.0.6/lib/asciidoctor/rfc/common/base.rb:15:in `convert'
```

**Comment by opoudjis** [comment]
There is a markup error in the example:

````
[example]
--
The following is an example of this property::

[source]
----
SXDATE:19960402T010000Z,19960403T010000Z,19960404T010000Z
----
--
````

That should be a single colon. The double colon means that Asciidoctor is trying to parse this as a definition list and failing. There are several instances of this erroneous markup in that document.

**Comment by opoudjis** [comment]
I do not have permissions to update this document. @douglm will need to do so.

**Comment by ronaldtse** [comment]
@opoudjis you now have access. I've corrected those issues but now running into a new error:

```sh
[metanorma-ietf] IDREF "RFC5545" without matching ID @ 974:73
[metanorma-ietf] IDREF "RFC7986" without matching ID @ 439:24
bundler: failed to load command: metanorma (/Users/me/.rbenv/versions/2.6.5/bin/metanorma)
JSON::ParserError: 767: unexpected token at ''
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/2.6.0/json/common.rb:156:in `parse'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/2.6.0/json/common.rb:156:in `parse'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-ietf-1.0.6/lib/asciidoctor/rfc/common/base.rb:430:in `block in cache_biblio'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-ietf-1.0.6/lib/asciidoctor/rfc/common/base.rb:429:in `open'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-ietf-1.0.6/lib/asciidoctor/rfc/common/base.rb:429:in `cache_biblio'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-ietf-1.0.6/lib/asciidoctor/rfc/common/base.rb:284:in `extract_entities'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-ietf-1.0.6/lib/asciidoctor/rfc/v2/base.rb:103:in `resolve_references'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-ietf-1.0.6/lib/asciidoctor/rfc/v2/base.rb:98:in `document'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-ietf-1.0.6/lib/asciidoctor/rfc/common/base.rb:15:in `convert'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/asciidoctor-1.5.8/lib/asciidoctor/document.rb:1190:in `convert'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/asciidoctor-1.5.8/lib/asciidoctor.rb:1521:in `convert'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-0.3.16/lib/metanorma/input/asciidoc.rb:8:in `process'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-ietf-1.0.6/lib/metanorma/ietf/processor.rb:33:in `input_to_isodoc'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-0.3.16/lib/metanorma/compile.rb:101:in `process_input'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-0.3.16/lib/metanorma/compile.rb:17:in `compile'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-cli-1.2.7/lib/metanorma/cli/compiler.rb:41:in `compile_file'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-cli-1.2.7/lib/metanorma/cli/compiler.rb:29:in `compile'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-cli-1.2.7/lib/metanorma/cli/compiler.rb:33:in `compile'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-cli-1.2.7/lib/metanorma/cli/command.rb:34:in `compile'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-cli-1.2.7/lib/metanorma/cli.rb:60:in `start'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/metanorma-cli-1.2.7/exe/metanorma:20:in `<top (required)>'
  /Users/me/.rbenv/versions/2.6.5/bin/metanorma:23:in `load'
```

**Comment by opoudjis** [comment]
I don't get those, and this sounds like a completely non-Metanorma issue (JSON). This is what I get:

````
[metanorma-ietf] IDREF "RFC5545" without matching ID @ 19:52
[metanorma-ietf] IDREF "RFC5545" without matching ID @ 62:103
[metanorma-ietf] IDREF "RFC5545" without matching ID @ 140:24
[metanorma-ietf] IDREF "RFC5545" without matching ID @ 192:85
[metanorma-ietf] IDREF "RFC5545" without matching ID @ 193:84
[metanorma-ietf] IDREF "RFC5545" without matching ID @ 194:86
[metanorma-ietf] IDREF "RFC5545" without matching ID @ 374:36
[metanorma-ietf] IDREF "RFC5545" without matching ID @ 817:24
[metanorma-ietf] IDREF "RFC5545" without matching ID @ 893:151
[metanorma-ietf] IDREF "RFC5545" without matching ID @ 901:25
[metanorma-ietf] IDREF "RFC5545" without matching ID @ 915:65
[metanorma-ietf] IDREF "RFC5545" without matching ID @ 952:69
[metanorma-ietf] IDREF "RFC5545" without matching ID @ 974:73
[metanorma-ietf] IDREF "RFC7986" without matching ID @ 439:24
Parsing file /var/folders/0x/fm2y9npj3jv50lx9xgwnnmr00000gs/T/draft-ietf-calext-icalendar-series.txt20191101-27796-igk9d2
Warning: No DTD given, defaulting to /usr/local/lib/python3.7/site-packages/xml2rfc/templates/rfc2629.dtd
Error: Unable to validate the XML document: /var/folders/0x/fm2y9npj3jv50lx9xgwnnmr00000gs/T/draft-ietf-calext-icalendar-series.txt20191101-27796-igk9d2
 <string>: Line 19: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 62: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 140: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 192: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 193: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 194: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 374: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 817: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 893: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 901: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 915: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 952: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 974: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 439: IDREF attribute target references an unknown ID "RFC7986"
Parsing file /var/folders/0x/fm2y9npj3jv50lx9xgwnnmr00000gs/T/draft-ietf-calext-icalendar-series.html20191101-27796-dg6r7w
Warning: No DTD given, defaulting to /usr/local/lib/python3.7/site-packages/xml2rfc/templates/rfc2629.dtd
Error: Unable to validate the XML document: /var/folders/0x/fm2y9npj3jv50lx9xgwnnmr00000gs/T/draft-ietf-calext-icalendar-series.html20191101-27796-dg6r7w
 <string>: Line 19: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 62: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 140: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 192: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 193: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 194: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 374: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 817: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 893: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 901: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 915: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 952: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 974: IDREF attribute target references an unknown ID "RFC5545"
 <string>: Line 439: IDREF attribute target references an unknown ID "RFC7986"
~/Documents/Arbeit/upwork/ribose/csd-icalendar-series[master] $ ls
````

**Comment by opoudjis** [comment]
I suspect you haven't checked a document in. I only have two.

**Comment by douglm** [comment]
I think this document was converted from the ietf XML document. I thought I'd used it to generate the last submitted doc but now I don't think so.
As such it was probably needing some work.

The subscription upgrade doc on the other hand was used some tiem back but doesn't build now. That one is actually the more pressing. I managed to submit the XML version of this one with the desired change (the name only to get it in the queue).

**Comment by opoudjis** [comment]
Unfortunately, @douglm, I don't understand the terms you are using: queue? I have zero context about this. All I can tell you is (a) there was a crash triggered by lists within examples (which it wasn't expecting, and that was a markup error); (b) I've updated the metanorma-ietf gem to deal with this; (c) @ronaldtse  refreshed the bibliography cache and got the documents compiling a few hours ago (put `:flush-cashes: true` temporarily in the document header); (d) I got the documents compiling as soon as I fixed the markup.

Because I have zero context about this: I need you to (a) do a bundle update to refresh the gems; (b) make the directory with flush-caches on; (c) if you still can't get this building, I need *detailed* logs from you, because I don't have any context for this work and what needs to build. (And, since I'm managing all of metanorma, I'm trying not to get any :-)

That includes which document is the subscription upgrade document. If it's cc-51003.adoc, I've had no problem building that.
